tests/go/boilerplate_test.go

```diff
@@ -0,0 +1,49 @@
+package _go
+
+import (
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func DecodeDeployment(t *testing.T, path string) *appsv1.Deployment {
+	subject := &appsv1.Deployment{}
+	scheme := NewSchemeWithDefault(t)
+	assert.NoError(t, appsv1.AddToScheme(scheme))
+	return DecodeWithSchema(t, path, subject, scheme).(*appsv1.Deployment)
+}
+
+func DecodeWithSchema(t *testing.T, path string, into runtime.Object, schema *runtime.Scheme) runtime.Object {
+	data, err := ioutil.ReadFile(path)
+	assert.NoError(t, err)
```

Comment:
Assert will not stop the test so following code will get executed. If not accounted for, this results in panics which make the test output hard to read. Ensure tests stop on errors where it does not make sense to continue.

```suggestion
	if err != nil {
		t.Fatalf("failed to read file: %w", err)
	}
```

Alternatively, use `testing.T.FailNow()`. See https://golang.org/pkg/testing/#T.FailNow.

Reply:
Using [require](https://github.com/stretchr/testify#require-package) seems to be the better solution?
```suggestion
	require.NoError(t, err)
```

Reply:
I was not aware of the `require` variants. TIL.

Reply:
I knew `require`, but didn't think of this during some refactoring into boilerplate. Boilerplate is the current name since we don't have the "test library" yet